### PR TITLE
Fix #3922: Hi fi tablet create profile rename fragment

### LINKED ISSUE #3922: Introduce ProfileRenameFragment
Creating a ProfileRenameFragment in the profile folder of the app module so that we can utilize the big tablet screen size by opening the fragment in a navhost fragment 
This issue is a sub-issue of #3241

### PR DESCRIPTION
**Pull request by yash10019coder**
## Explanation

Fixes #3922 
Created the the fragment and its layout
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

| Old | New |
| --------------------- | -------------------------|
| Screenshot_20211128_133412 | mobile_port |
| Screenshot_20211128_133415 | mobile_land |
| Screenshot_20211128_133217 | tab_port |
| Screenshot_20211128_133223 | tab_land |

- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))

image
image

### CODE REVIEW

**Comment by rt4914** [comment]
@yash10019coder Please follow these steps whenever PR is ready for review: https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

**Comment by BenHenning** [comment]
Is this ready for review @yash10019coder?

**Comment by yash10019coder** [comment]
No It isn't ready yet , I will sure do it at the end of the Oct

**Comment by BenHenning** [comment]
Sounds good, thanks @yash10019coder! In that case, I'm going to convert this to a 'Draft' so that it's clearer that this isn't yet finished. You can convert it back to being a non-Draft once it's ready.

**Comment by yash10019coder** [comment]
Hi getting this error @rt4914 
image
please help 
`AAPT: /home/ubuntu/opensource/oppia-android/app/build/intermediates/incremental/mergeDebugResources/stripped.dir/layout-sw600dp-v13/profile_progress_header.xml:19: error: resource dimen/profile_edit_image_layout_margin_start (aka org.oppia.android:dimen/profile_edit_image_layout_margin_start) not found.
/home/ubuntu/opensource/oppia-android/app/build/intermediates/incremental/mergeDebugResources/stripped.dir/layout-sw600dp-v13/profile_progress_header.xml:45: error: resource dimen/profile_name_text_view_layout_margin_end (aka org.oppia.android:dimen/profile_name_text_view_layout_margin_end) not found.
/home/ubuntu/opensource/oppia-android/app/build/intermediates/incremental/mergeDebugResources/stripped.dir/layout-sw600dp-v13/profile_progress_header.xml:153: error: resource dimen/recently_played_stories_text_view_layout_margin_start (aka org.oppia.android:dimen/recently_played_stories_text_view_layout_margin_start) not found.
/home/ubuntu/opensource/oppia-android/app/build/intermediates/incremental/mergeDebugResources/stripped.dir/layout-sw600dp-v13/profile_progress_header.xml:169: error: resource dimen/view_all_text_view_margin_end (aka org.oppia.android:dimen/view_all_text_view_margin_end) not found.
error: failed linking file resources.`
thanks

**Comment by rt4914** [comment]
@yash10019coder Unassigning as Veena has provided you reference to implement this PR.

**Comment by yash10019coder** [comment]
hi @rt4914 here's the video 

https://user-images.githubusercontent.com/76404817/139912779-b39a2dfe-a93a-4c97-a56a-6b5dd6bb33cc.mp4

**Comment by rt4914** [comment]
@yash10019coder The video is not playable.
Also the error for me is in landscape mode only because `land/profile-rename-activity.xml` file has not been updated similar to its other file.

You will need to update the code in `land/profile-rename-activity.xml` file too.

Also, if you are getting a different error then you can you mention the steps that you are following.

**Comment by BenHenning** [comment]
FYI I'm listed as a codeowner for this PR & I'll be unavailable to perform code reviews over the next 2 weeks--thanks for your patience.

**Review by yash10019coder — COMMENTED** [review]
Hi @rt4914 added all the Kdocs and added ProfileRenameFragment and its Presenter in exempted test files PTAL.

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragment.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.oppia.android.app.fragment.InjectableFragment
+import javax.inject.Inject
+
+/** Fragment that provides functionality for renaming the profile name. */
+class ProfileRenameFragment : InjectableFragment() {
+  @Inject
+  lateinit var profileRenameFragmentPresenter: ProfileRenameFragmentPresenter
+
+  companion object {
+    private const val PROFILE_ID = "Profile_Id"
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragment.kt]: Please rename it correctly, something like
PROFILE_ID_ARGUMENT_KEY = "ProfileRenameFragment.profile_id"

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.utility.TextInputEditTextHelper.Companion.onTextChanged
+import org.oppia.android.app.viewmodel.ViewModelProvider
+import org.oppia.android.databinding.ProfileRenameFragmentBinding
+import org.oppia.android.domain.profile.ProfileManagementController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+/** The Presenter for [ProfileRenameFragment]. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt]: Change `Presenter` -> `presenter`

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  private val viewModelProvider: ViewModelProvider<ProfileRenameViewModel>,
+  private val resourceHandler: AppLanguageResourceHandler
+) {
+  private val renameViewModel: ProfileRenameViewModel by lazy {
+    getProfileRenameViewModel()
+  }
+
+  private lateinit var binding: ProfileRenameFragmentBinding
+
+  /** Handlers onCreateView() method of [ProfileRenameFragment]. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt]: Nit: `Handlers` -> `Handles`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Took overview look. Please create `ProfileRenameFragmentTest` file and transfer correct tests from `ProfileRenameActivityTest` to this new file. (You might need to change test names a bit like changing `Activity` to `Fragment` at various places.)

Overall its really nice.

**Comment by rt4914** [comment]
@yash10019coder Also add before and after screenshot in mobile portrait/landscape so as to demonstrate that UI does not change at all.

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragment.kt (hunk 1 above)
Comment (yash10019coder) [reply app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragment.kt]: done

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt (hunk 2 above)
Comment (yash10019coder) [reply app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt]: done

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt (hunk 3 above)
Comment (yash10019coder) [reply app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt]: done

**Review by yash10019coder — COMMENTED** [review]
Hi @rt4914 PTAL

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+
+  @Test
+  fun testProfileRenameFragment_clickSave_changeConfiguration_saveButtonIsNotClickable() {
+    ActivityScenario.launch<ProfileRenameActivity>(
+      ProfileRenameActivity.createProfileRenameActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      Espresso.onView(ViewMatchers.withId(R.id.profile_rename_save_button))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: Please use direct imports everywhere
Example
`onView(withId(R.id.profile_rename_save_button)).check(matches(not(isClickable())))`

This makes code more readable.

**Review by rt4914 — CHANGES_REQUESTED** [review]
The landscape mode screenshots have changed and that means the UI needs to be fixed such that before/after screenshots are same.

Otherwise really nice work.

**Comment by yash10019coder** [comment]
> The landscape mode screenshots have changed and that means the UI needs to be fixed such that before/after screenshots are same.
> 
> Otherwise really nice work.

yeah I noticed it now I think that adding a padding will make it same

**Review by yash10019coder — COMMENTED** [review]
Hi @rt4914 mentioned all the comments PTAL

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt (hunk 4 above)
Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: fixed

**Comment by yash10019coder** [comment]
> The landscape mode screenshots have changed and that means the UI needs to be fixed such that before/after screenshots are same.
> 
> Otherwise really nice work.

fixed

**Comment by BenHenning** [comment]
Hi. FYI I've been out the last couple of weeks, so I'm working to catch up on my reviews. I might be delayed a couple of days, but I'll be reviewing this soon. Thanks for your patience!

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+      )
+    ).use {
+      onView(
+        allOf(
+          withId(R.id.profile_rename_input_edit_text),
+          isDescendantOfA(withId(R.id.profile_rename_input))
+        )
+      ).perform(editTextInputAction.appendText("James"))
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_save_button))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: This should be in single line
`onView(withId(R.id.profile_rename_save_button)).perform(click())`

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+        allOf(
+          withId(R.id.profile_rename_input_edit_text),
+          isDescendantOfA(withId(R.id.profile_rename_input))
+        )
+      ).perform(editTextInputAction.appendText("James"))
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_save_button))
+        .check(matches(isEnabled()))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
One more change, in test cases a lot of multiple lines code can be reduced to single line. I have left 2 comments but check for them throughout the file. Thanks.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt (hunk 5 above)
Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt (hunk 6 above)
Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: done

**Review by yash10019coder — COMMENTED** [review]
Hi @rt4914 @BenHenning made and commented all the requested changes PTAL

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt [hunk 7]
```diff
@@ -0,0 +1,480 @@
...
+        allOf(
+          withId(R.id.profile_rename_input_edit_text),
+          isDescendantOfA(withId(R.id.profile_rename_input))
+        )
+      ).perform(editTextInputAction.appendText("Admin"))
+      onView(withId(R.id.profile_rename_save_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_input)).check(
+        matches(
+          TextInputAction.hasErrorText(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: Use direct import for this one.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt [hunk 8]
```diff
@@ -0,0 +1,480 @@
...
+          isDescendantOfA(withId(R.id.profile_rename_input))
+        )
+      ).perform(editTextInputAction.appendText("123"))
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_save_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_input))
+        .check(
+          matches(
+            TextInputAction.hasErrorText(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: Ditto

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt [hunk 9]
```diff
@@ -0,0 +1,480 @@
...
+      )
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_save_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_rename_input))
+        .check(
+          matches(
+            TextInputAction.hasErrorText(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
@yash10019coder PTAL

**Comment by rt4914** [comment]
@yash10019coder Merge with latest develop too.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt (hunk 7 above)
Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt (hunk 8 above)
Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt (hunk 9 above)
Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt]: done

**Review by BenHenning — APPROVED** [review]
Approved for codeowners (only had 2 files to check).

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**